### PR TITLE
Fix missing cpu/memory cgroups for the systemd services on CentOS

### LINKED ISSUE #440: CentOS kubelet complains about systemd slices
We should probably set `--runtime-cgroups=/systemd/system.slice --kubelet-cgroups=/systemd/system.slice` as kubelet args on CentOS/RHEL.

### PR DESCRIPTION
**Pull request by SpComb**
Fixes #440

Enable `CPUAccounting` and `MemoryAccounting` for the `kubelet.service`, allowing systemd to create per-service cpu/memory cgroups for all `system.slice` services.

### CODE REVIEW

File: lib/pharos/host/el7/scripts/ensure-kubelet.sh [hunk 1]
```diff
@@ -6,13 +6,25 @@
...
-    cat <<EOF >/etc/systemd/system/kubelet.service.d/05-pharos-kubelet.conf
+
+cat <<EOF >/etc/systemd/system/kubelet.service.d/05-pharos-kubelet.conf
 [Service]
 ExecStartPre=-/sbin/swapoff -a
 ExecStart=
 ExecStart=/usr/bin/kubelet ${KUBELET_ARGS} --cgroup-driver=systemd --pod-infra-container-image=${IMAGE_REPO}/pause-${ARCH}:3.1
 EOF
 
+cat <<EOF >/etc/systemd/system/kubelet.service.d/11-cgroups.conf
```
Comment (SpComb) [inline lib/pharos/host/el7/scripts/ensure-kubelet.sh]: Put this here to make it CentOS-only... however, I think we we could also just include it in the `11-pharos.conf` for all hosts... might be clearer. I don't think this should have any side-effects... it should match the default systemd behavior on Ubuntu?

**Comment by SpComb** [comment]
Testing on CentOS:

```
[root@terom-centos-test ~]# systemctl cat kubelet.service
# /etc/systemd/system/kubelet.service
[Unit]
Description=kubelet: The Kubernetes Node Agent
Documentation=http://kubernetes.io/docs/

[Service]
ExecStart=/usr/bin/kubelet
Restart=always
StartLimitInterval=0
RestartSec=10

[Install]
WantedBy=multi-user.target

# /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
[Service]
Environment="KUBELET_KUBECONFIG_ARGS=--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf"
Environment="KUBELET_SYSTEM_PODS_ARGS=--pod-manifest-path=/etc/kubernetes/manifests --allow-privileged=true"
Environment="KUBELET_NETWORK_ARGS=--network-plugin=cni --cni-conf-dir=/etc/cni/net.d --cni-bin-dir=/opt/cni/bin"
Environment="KUBELET_DNS_ARGS=--cluster-dns=10.96.0.10 --cluster-domain=cluster.local"
Environment="KUBELET_AUTHZ_ARGS=--authorization-mode=Webhook --client-ca-file=/etc/kubernetes/pki/ca.crt"
Environment="KUBELET_CADVISOR_ARGS=--cadvisor-port=0"
Environment="KUBELET_CGROUP_ARGS=--cgroup-driver=systemd"
Environment="KUBELET_CERTIFICATE_ARGS=--rotate-certificates=true --cert-dir=/var/lib/kubelet/pki"
ExecStart=
ExecStart=/usr/bin/kubelet $KUBELET_KUBECONFIG_ARGS $KUBELET_SYSTEM_PODS_ARGS $KUBELET_NETWORK_ARGS $KUBELET_DNS_ARGS $KUBELET_AUTHZ_ARGS $KUBELET_CADVISOR_ARGS $KUBELET_CGROUP_ARGS $KUBELET_CERTIFICATE_ARGS $KUBELET_EXTRA_ARGS
# /etc/systemd/system/kubelet.service.d/11-cgroups.conf
[Service]

# https://github.com/kontena/pharos-cluster/issues/440
# the kubelet expects to find cpu+memory cgroups for the kubelet and docker services in order to report system container stats
# have systemd create per-service cpu+memory cgroups for the kubelet.service
# as a side effect, this will also create cpu+memory cgroups for all system.slice services, including docker.service
CPUAccounting=true
MemoryAccounting=true
# /etc/systemd/system/kubelet.service.d/11-pharos.conf
[Service]
Environment='KUBELET_EXTRA_ARGS=--read-only-port=0 --node-ip=167.99.139.236 --hostname-override=terom-centos-test --pod-infra-container-image=quay.io/kontena/pause-amd64:3.1'
Environment='KUBELET_DNS_ARGS=--cluster-dns=10.250.0.10 --cluster-domain=cluster.local'
ExecStartPre=-/sbin/swapoff -a
```

```
[root@terom-centos-test ~]# cat /proc/$(pidof kubelet)/cgroup
11:freezer:/
10:net_prio,net_cls:/
9:memory:/system.slice/kubelet.service
8:pids:/
7:perf_event:/
6:blkio:/
5:cpuset:/
4:devices:/
3:hugetlb:/
2:cpuacct,cpu:/system.slice/kubelet.service
1:name=systemd:/system.slice/kubelet.service
```

```
[root@terom-centos-test ~]# journalctl -u kubelet | grep 'Failed to get system container stats'
[root@terom-centos-test ~]# 
```

**Comment by SpComb** [comment]
Alternatively, this could be fixed directly in the kubelet RPM.

**Comment by jakolehm** [comment]
Fixed in kubelet rpm package.

**Comment by jakolehm** [comment]
@washingtoneg Pharos 1.2.0 kubelet rpm package has this fix (see https://dl.bintray.com/kontena/pharos-rpm/x86_64/kubelet-1.10.4-1.x86_64.rpm)... rpm build/spec repo is not (yet) public.

**Pull request closed without merging**